### PR TITLE
Return azure core HttpResponseException for service side exceptions

### PR DESCRIPTION
**Pull request by samvaity**
Simplify `ErrorResponseException` model to return `ErrorInformation` directly instead of `ErrorResponse`

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientTest.java [hunk 1]
```diff
@@ -167,8 +170,12 @@ public void recognizeReceiptDataTextDetails(HttpClient httpClient, FormRecognize
     @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
     public void recognizeReceiptInvalidSourceUrl(HttpClient httpClient, FormRecognizerServiceVersion serviceVersion) {
         client = getFormRecognizerAsyncClient(httpClient, serviceVersion);
-        invalidSourceUrlRunner((sourceUrl) -> assertThrows(ErrorResponseException.class,
-            () -> client.beginRecognizeReceiptsFromUrl(sourceUrl).getSyncPoller()));
+        invalidSourceUrlRunner((sourceUrl) -> {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientTest.java:173]: sourceUrl -> invalidSourceUrl

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java [hunk 2]
```diff
@@ -187,4 +190,21 @@ Pair readBytes(int cnt) {
...
     }
+
+    /**
+     * Mapping a {@link ErrorResponseException} to {@link HttpResponseException} if exist. Otherwise, return
+     * original {@link Throwable}.
+     *
+     * @param throwable A {@link Throwable}.
+     * @return A {@link HttpResponseException} or the original throwable type.
+     */
+    public static Throwable mapToHttpResponseExceptionIfExist(Throwable throwable) {
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java:201]: @srnagar Does this seem like something we would want the autorest to be doing for the client ?

Comment (sima-zhu) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java:201]: Yes, I think it is better to define in swagger or make it configurable on swagger readme. Search has the same problem.

Comment (srnagar) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java:201]: AutoRest will always provide the most specific error response. It's for the convenience layer to determine if it should be converted to a more generic exception type.

**Pull request merged**